### PR TITLE
feat: add support for explain prepared query API endpoint

### PR DESCRIPTION
**Pull request by larrytamnjong**
Add support for Explain Prepared Query API endpoint in Consul.Net (implementation only, tests pending)
#367 
@naskio

### CODE REVIEW

**Comment by naskio** [comment]
Hi @larrytamnjong,
As a best practice, I suggest keeping this PR as a draft while it's still in progress (for example, when tests are missing)
Thanks

**Comment by larrytamnjong** [comment]
Sure👍, thanks for the feedback.

**Comment by larrytamnjong** [comment]
@naskio I've added the tests. Please take a look when you have time. Thanks!

**Review by marcin-krystianc — COMMENTED** [review]
Hi @larrytamnjong, thanks for the PR. 
I've left just a single comment to remove unnecessary lines, it looks good otherwise.

File: Consul.Test/PreparedQueryTest.cs [hunk 1]
```diff
@@ -109,6 +109,19 @@ public async Task PreparedQuery_Test()
             Assert.True(nodes.Length == 1);
             Assert.Equal("foobaz", results.Nodes[0].Node.Name);
 
+            var query = (await _client.PreparedQuery.Explain(id)).Response.Query;
+
+            Assert.NotNull(query);
+            Assert.True(query.Name == "my-query");
+            Assert.True(query.Service.OnlyPassing == true);
+
+            query = null;
```
[the remark this answers is not in the document]
Comment (larrytamnjong) [reply Consul.Test/PreparedQueryTest.cs:118]: @marcin-krystianc thanks for the feedback let me do that..

**Comment by larrytamnjong** [comment]
@marcin-krystianc  all the null assignments have been removed.

**Review by marcin-krystianc — APPROVED** [review]
Nice👍 Thank you!

**Pull request merged**